File: js/examples/mcp_conversion_example.js
#!/usr/bin/env node
/**
 * Ejemplo de uso del convertidor MCP a ATDF en JavaScript.
 * 
 * Este script demuestra cómo utilizar el módulo MCPConverter.js para transformar
 * herramientas desde el formato MCP al formato ATDF.
 */

const path = require('path');
const fs = require('fs');
const { mcpToAtdf, batchConvert } = require('../src/MCPConverter');

// Crear directorio para los resultados
const outputDir = path.join(__dirname, 'output');
if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

console.log('🔄 Iniciando ejemplo de conversión MCP a ATDF en JavaScript...');

// Ejemplo 1: Herramienta simple convertida programáticamente
console.log('\n==== Ejemplo 1: Conversión programática simple ====');

const mcpToolExample = {
    name: 'fetch',
    description: 'Recupera contenido de una URL',
    annotations: {
        context: 'Cuando necesites obtener datos de una página web'
    },
    inputSchema: {
        type: 'object',
        properties: {
            url: {
                type: 'string',
                description: 'URL a recuperar'
            },
            raw: {
                type: 'boolean',
                description: 'Devolver HTML crudo (opcional)'
            }
        },
        required: ['url']
    }
};

try {
    // Convertir a formato básico
    const basicResult = mcpToAtdf(mcpToolExample);
    const basicOutputPath = path.join(outputDir, 'fetch_basic.json');
    fs.writeFileSync(basicOutputPath, JSON.stringify(basicResult, null, 2));
    console.log(`✅ Herramienta básica guardada en: ${basicOutputPath}`);
    
    // Convertir a formato mejorado
    const enhancedResult = mcpToAtdf(mcpToolExample, { 
        enhanced: true, 
        author: 'MCP Example JS' 
    });
    const enhancedOutputPath = path.join(outputDir, 'fetch_enhanced.json');
    fs.writeFileSync(enhancedOutputPath, JSON.stringify(enhancedResult, null, 2));
    console.log(`✅ Herramienta mejorada guardada en: ${enhancedOutputPath}`);
    
    // Mostrar resultado
    console.log('\nContenido del archivo ATDF generado:');
    console.log('-'.repeat(50));
    console.log(JSON.stringify(basicResult, null, 2));
    console.log('-'.repeat(50));
} catch (error) {
    console.error(`❌ Error: ${error.message}`);
}

// Ejemplo 2: Procesamiento por lotes
console.log('\n==== Ejemplo 2: Procesamiento por lotes ====');

// Crear múltiples herramientas para el ejemplo
const mcpTools = {
    tools: [
        {
            name: 'search',
            description: 'Busca información en la web',
            annotations: {
                purpose: 'Obtener resultados de búsqueda de un término'
            },
            inputSchema: {
                type: 'object',
                properties: {
                    query: { type: 'string', description: 'Término de búsqueda' },
                    limit: { type: 'number', description: 'Número máximo de resultados' }
                },
                required: ['query']
            }
        },
        {
            name: 'weather',
            description: 'Obtiene información meteorológica',
            annotations: {
                context: 'Para conocer las condiciones climáticas de una ubicación'
            },
            inputSchema: {
                type: 'object',
                properties: {
                    location: { type: 'string', description: 'Ciudad o coordenadas' },
                    units: { type: 'string', description: 'Unidades (metric/imperial)' }
                },
                required: ['location']
            }
        }
    ]
};

// Guardar herramientas MCP en un archivo temporal
const mcpFilePath = path.join(outputDir, 'mcp_tools.json');
fs.writeFileSync(mcpFilePath, JSON.stringify(mcpTools, null, 2));

// Procesar por lotes
const batchOutputDir = path.join(outputDir, 'batch');
try {
    // Convertir todas las herramientas
    const results = batchConvert(mcpTools, batchOutputDir, { enhanced: true });
    console.log(`\n🎉 Conversión por lotes completada. Se convirtieron ${results.length} herramientas.`);
} catch (error) {
    console.error(`❌ Error en el procesamiento por lotes: ${error.message}`);
}

console.log(`\n🎉 Conversión completa. Resultados guardados en: ${outputDir}`); 

File: examples/mcp_conversion_example.py
#!/usr/bin/env python3
"""
Ejemplo de uso del convertidor MCP a ATDF.

Este script demuestra cómo utilizar el convertidor para transformar
herramientas desde el formato MCP al formato ATDF, tanto básico como mejorado.
"""

import json
import os
import sys
from pathlib import Path

# Añadir el directorio raíz al path para importar los módulos
sys.path.insert(0, str(Path(__file__).parent.parent))

from tools.converter import save_tool

# Importar el convertidor
from tools.mcp_converter import convert_mcp_file, mcp_to_atdf

# Crear directorio para los resultados
output_dir = Path(__file__).parent / "output"
output_dir.mkdir(exist_ok=True)


def main():
    """Ejemplo principal de conversión MCP a ATDF."""

    print("🔄 Iniciando ejemplo de conversión MCP a ATDF...")

    # Ejemplo 1: Herramienta simple convertida programáticamente
    print("\n==== Ejemplo 1: Conversión programática simple ====")

    mcp_tool_example = {
        "name": "fetch",
        "description": "Recupera contenido de una URL",
        "annotations": {"context": "Cuando necesites obtener datos de una página web"},
        "inputSchema": {
            "type": "object",
            "properties": {
                "url": {"type": "string", "description": "URL a recuperar"},
                "raw": {
                    "type": "boolean",
                    "description": "Devolver HTML crudo (opcional)",
                },
            },
            "required": ["url"],
        },
    }

    # Convertir a formato básico
    basic_result = mcp_to_atdf(mcp_tool_example)
    basic_output_path = output_dir / "fetch_basic.json"
    save_tool(basic_result, basic_output_path)
    print(f"✅ Herramienta básica guardada en: {basic_output_path}")

    # Convertir a formato mejorado
    enhanced_result = mcp_to_atdf(mcp_tool_example, enhanced=True, author="MCP Example")
    enhanced_output_path = output_dir / "fetch_enhanced.json"
    save_tool(enhanced_result, enhanced_output_path)
    print(f"✅ Herramienta mejorada guardada en: {enhanced_output_path}")

    # Ejemplo 2: Múltiples herramientas MCP en un solo archivo
    print("\n==== Ejemplo 2: Procesamiento por lotes ====")

    # Crear un archivo MCP temporal para el ejemplo
    mcp_tools = {
        "tools": [
            {
                "name": "search",
                "description": "Busca información en la web",
                "annotations": {
                    "purpose": "Obtener resultados de búsqueda de un término"
                },
                "inputSchema": {
                    "type": "object",
                    "properties": {
                        "query": {
                            "type": "string",
                            "description": "Término de búsqueda",
                        },
                        "limit": {
                            "type": "number",
                            "description": "Número máximo de resultados",
                        },
                    },
                    "required": ["query"],
                },
            },
            {
                "name": "weather",
                "description": "Obtiene información meteorológica",
                "annotations": {
                    "context": "Para conocer las condiciones climáticas de una ubicación"
                },
                "inputSchema": {
                    "type": "object",
                    "properties": {
                        "location": {
                            "type": "string",
                            "description": "Ciudad o coordenadas",
                        },
                        "units": {
                            "type": "string",
                            "description": "Unidades (metric/imperial)",
                        },
                    },
                    "required": ["location"],
                },
            },
        ]
    }

    # Guardar herramientas MCP en un archivo temporal
    mcp_file_path = output_dir / "mcp_tools.json"
    with open(mcp_file_path, "w", encoding="utf-8") as f:
        json.dump(mcp_tools, f, indent=2)

    # Procesar cada herramienta en el archivo
    batch_output_dir = output_dir / "batch"
    batch_output_dir.mkdir(exist_ok=True)

    for i, tool in enumerate(mcp_tools["tools"]):
        # Convertir y guardar cada herramienta
        atdf_tool = mcp_to_atdf(tool, enhanced=True)
        tool_path = batch_output_dir / f"{tool['name']}.json"
        save_tool(atdf_tool, tool_path)
        print(f"✅ Herramienta {i+1}/{len(mcp_tools['tools'])} convertida: {tool_path}")

    print("\n🎉 Conversión completada exitosamente!")
    print(f"📁 Resultados guardados en: {output_dir}")


if __name__ == "__main__":
    main()
